### PR TITLE
fixes #3190 - Fix STI auditable_type - Audit should say updated Location/Organization rather than updated Taxonomy and ComputeResource should not use STI class

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: app/models/concerns/audit_extensions.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def ensure_username
   def fix_auditable_type
     self.auditable_type = "Host"         if self.auditable_type == "Puppet::Rails::Host"
     self.associated_type = "Host"        if self.associated_type == "Puppet::Rails::Host"
-    self.auditable_type = auditable.type if self.auditable_type == "ComputeResource"
+    self.auditable_type = auditable.type if ["ComputeResource", "Taxonomy"].include?(self.auditable_type)
```
Comment (ohadlevy) [inline app/models/concerns/audit_extensions.rb]: looks good, couple of questions:
1. do we have the same issues with other sti ? (settings, hosts)
2. can we fix for CR and remove the Foreman::Model::ComputeResource?
3. should we not have a migration to fix the old audits? is it possible in this case?

also, in the above lines 41-41, we no longer have Puppet::Rails::Host

**Comment by isratrade** [comment]
- removed references to Puppet::Rails::Host
- added migration to fix auditable_type (note, if the taxonomy was deleted, the type can't be updated).

File: app/models/concerns/audit_extensions.rb [hunk 2]
```diff
@@ -38,12 +38,11 @@ def ensure_username
   end
 
   def fix_auditable_type
-    self.auditable_type = "Host"         if self.auditable_type == "Puppet::Rails::Host"
-    self.associated_type = "Host"        if self.associated_type == "Puppet::Rails::Host"
-    self.auditable_type = auditable.type if self.auditable_type == "ComputeResource"
     # STI Host class should use the stub module instead of Host::Base
     self.auditable_type = "Host"         if self.auditable_type =~  /Host::/
     self.associated_type = "Host"        if self.associated_type =~ /Host::/
+    self.auditable_type = auditable.type  if ["ComputeResource", "Taxonomy"].include?(self.auditable_type)
```
Comment (isratrade) [inline app/models/concerns/audit_extensions.rb]: I think I should remove ComputeResource. currently, this is what is shows
`Update Foreman/Model/Ec2: ec2-eu (eu-west-1-EC2)`
I prefer the parent class Compute Resource
`Update Compute Resource: ec2-eu (eu-west-1-EC2)`
Do you agree? @ohadlevy

**Comment by isratrade** [comment]
There is not an issue with settings, since the default is the parent class, so it shows
`updated Setting`
which is correct rather than
`updated Setting::General`

File: app/models/concerns/audit_extensions.rb (hunk 2 above)
Comment (ohadlevy) [inline app/models/concerns/audit_extensions.rb]: :+1:

**Comment by isratrade** [comment]
added ComputeResource to migration and removed it from audit_extensions so parent class is used as auditable_type

File: db/migrate/20131003143143_fix_auditable_type2.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class FixAuditableType2 < ActiveRecord::Migration
+  def up
+    Audit.where(:auditable_type => 'Taxonomy').each do |audit|
+      taxonomy_type = Taxonomy.find_by_id(audit.auditable_id).try(:type)
+      audit.update_attribute(:auditable_type, taxonomy_type) if taxonomy_type
+    end
+
+    Audit.where(:auditable_type => ['Foreman::Model::Ovirt',
```
Comment (ohadlevy) [inline db/migrate/20131003143143_fix_auditable_type2.rb]: can we use a like statement instead? or maybe use the constant from the CR class?

Comment (isratrade) [inline db/migrate/20131003143143_fix_auditable_type2.rb]: I originally used a constant for the error, but Rails gave an error. I can generate a SQL LIKE statement %Foreman::Model% but this syntax might only be for postgres, not sure.

Comment (ohadlevy) [inline db/migrate/20131003143143_fix_auditable_type2.rb]: Use scope search if you are not sure?:-)

**Comment by isratrade** [comment]
@ohadlevy, re-committed with LIKE

**Comment by domcleal** [comment]
Merged as 409d7f1, thanks @isratrade!

**Pull request closed without merging**